plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/Meta.kt

```diff
@@ -21,3 +21,11 @@ data class WithPosition(override val inner: ExpEmbedding, val source: KtSourceEl
         get() = inner.debugTreeView
 }
 
+fun ExpEmbedding.withPosition(source: KtSourceElement?): ExpEmbedding =
+    when {
+        // Inner position is more specific anyway
+        this is WithPosition -> this
+        source == null -> this
```

Comment:
Same thing here, I think you can inline them.

Reply:
I'd rather keep it this way since the comment only applies to the first branch, and the rhs is simple anyway.